src/main/java/org/openjdk/jextract/impl/StructBuilder.java

```diff
@@ -81,60 +88,71 @@ private List<String> prefixNamesList() {
         return Collections.unmodifiableList(prefixes);
     }
 
-    @Override
-    void classBegin() {
-        if (!inAnonymousNested()) {
-            super.classBegin();
-            layoutConstant = constants().addLayout(((Type.Declared) structType).tree().layout().orElseThrow());
-            layoutConstant.emitGetter(this, MEMBER_MODS, Constant::nameSuffix);
-        }
+    private boolean isNested() {
+        return !enclosingClassNames().isEmpty();
     }
 
-    @Override
-    void classDeclBegin() {
+    void begin() {
         if (!inAnonymousNested()) {
+            if (isNested()) {
+                sourceFileBuilder().incrAlign();
+            }
             emitDocComment(structTree);
+            classBegin();
+            Constant layoutConstant = constants.addLayout(((Type.Declared) structType).tree().layout().orElseThrow());
+            layoutConstant.emitGetter(this, MEMBER_MODS, Constant::nameSuffix);
```

Comment:
Note that I didn't get rid of _all_ the context dependent logic in this patch. That would require more surgery in OutputFactory, which I've left alone for now.